### ISSUE TITLE
do we need an interface to fill forms, click, etc?

### ISSUE THREAD

**Issue by inf3rno** [inf3rno/e2e-testing#14]
In #5 we already discussed that nightwatch has its own interface to handle user input. So for example it fills forms, clicks on links, etc. We need to check what actually nightwatch and selenium and phantomjs supports and think about whether we need to support that too, or it is just enough to do it with jquery or another DOM lib. We need to check how hard it is with vanilla js too... So this is about usability, and not about necessity...

**Comment by inf3rno** [comment]
I think this could be a plugin and it should be optional. Or maybe a default plugin, which can be turned off.

**Comment by inf3rno** [comment]
I think we need an action lib for firing events too. An user action results typically in a bunch of events, not just a single event. For example a click is `mousemove, mousedown, mouseup, click`, typing is `keydown, keyup, keypress, change, input`, etc... Firing a single event to simulate user input is just plain wrong. I haven't found any lib, which does something like this. Most of them is something like `fire(type, options)` or even worse. We need a very basic user interaction API for now, and later we can extend it.

**Comment by inf3rno** [comment]
I think we already answered this question in #71 and #77. We will need to connect the link following and form sending to the lib and because of this we need an API at least with 2 methods, which return the proper promises after following links or sending forms. I already added this I think in the #77 in a commit. So the work is done. Later we can extend these methods or move them from the navigator to an interactor as we originally planned. For now they are ok in the navigator, because they are not that complex.
